### PR TITLE
Add p2 user management UI improvements

### PR DESCRIPTION
**Pull request by naxoc**
#### Changes proposed in this Pull Request
 

This PR changes a bunch of strings for P2s in the user management area to make it easier to manage P2 users and guests.

Specifically it:
* Adds site name to some headers to make it easier to know if user management is on the workspace or the P2.
* Removes all redirects for user management from the P2 (space) to the workspace (hub) because we now support the guest concept in P2.
* Adds a guest count on P2s that have guests
* Adds a guest badge to users that are guests on the P2
* Adds a react-query `useP2GuestsQuery` to ask the API for guests (needs PR 3586-gh-Automattic/p2 to be merged)
* Adds a HOC wrapping the guest query
* Removes a hard restriction that disabled banners for all P2s. Banners are now allowed if they have the `p2-banner` class.

#### Testing instructions

See 3405-gh-Automattic/p2 for testing instructions

### CODE REVIEW

File: client/my-sites/people/people-profile/index.jsx [hunk 1]
```diff
@@ -219,8 +223,26 @@ const PeopleProfile = ( { siteId, type, user, invite } ) => {
 				</>
 			);
 		}
+		if ( p2Guests?.guests && p2Guests.guests.includes( user?.linked_user_ID ?? user?.ID ) ) {
+			p2GuestBadge = (
+				<>
+					<div className="people-profile__role-badge role-p2-guest">
+						{ translate( 'Guest', {
```
Comment (a8ci18n) [inline client/my-sites/people/people-profile/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **81** times:
`translate( 'Guest' )` *ES Score: 12*


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/people/team-list/team.jsx [hunk 2]
```diff
@@ -22,6 +26,40 @@ class Team extends React.Component {
...
+			translateOptions.args.numberGuests = p2Guests?.found;
+			return translate(
+				'There is %(numberPeople)d member and %(numberGuests)d guests in this P2',
+				'There are %(numberPeople)d members and %(numberGuests)d guests in this P2',
+				translateOptions
+			);
+		}
+
+		return translate(
+			'There is %(numberPeople)d members in this P2',
```
Comment (annemirasol) [inline client/my-sites/people/team-list/team.jsx]: extra s in "members" for the singular version.

File: client/my-sites/people/team-list/team.jsx [hunk 3]
```diff
@@ -22,6 +26,40 @@ class Team extends React.Component {
...
+		if ( isP2HubSite ) {
+			return translate(
+				'There is %(numberPeople)d members in this workspace',
+				'There are %(numberPeople)d members in this workspace',
+				translateOptions
+			);
+		}
+
+		if ( p2Guests?.found ) {
+			translateOptions.args.numberGuests = p2Guests?.found;
```
Comment (annemirasol) [inline client/my-sites/people/team-list/team.jsx]: Nit: the optional chain is no longer needed here, I think.

**Review by annemirasol — APPROVED** [review]
Works great, no regressions noted.

Good job, Camilla!

File: client/my-sites/people/team-list/team.jsx [hunk 4]
```diff
@@ -22,6 +26,40 @@ class Team extends React.Component {
...
+			return translate(
+				'There is %(numberPeople)d member and %(numberGuests)d guests in this P2',
+				'There are %(numberPeople)d members and %(numberGuests)d guests in this P2',
```
Comment (annemirasol) [inline client/my-sites/people/team-list/team.jsx]: Not sure if this works the way I understand it... but will this cover all the correct pluralization combinations? I notice it's "guests" for both strings, but 'numberGuests' is being passed.

Comment (naxoc) [reply client/my-sites/people/team-list/team.jsx]: You're right. That would be off if there is say 1 member and 3 guests. I pushed a change in fc2944a8942bd74aed26fdac2a538dec9db6f91e that also makes the guests singular/plural. I'm not 100% that the approach will work for all languages. What do you think?

File: client/my-sites/people/team-list/team.jsx [hunk 5]
```diff
@@ -22,6 +26,54 @@ class Team extends React.Component {
...
+		const guests = translate( '%(numberGuests)d guest', '%(numberGuests)d guests', {
+			args: {
+				numberGuests: p2Guests.found,
+			},
+			count: p2Guests.found,
+		} );
+
+		const numMembers = totalUsers - p2Guests.found;
+		return translate(
+			'There is %(numMembers)d member and %(guests)s in this P2',
```
Comment (a8ci18n) [inline client/my-sites/people/team-list/team.jsx:64]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/people/team-list/team.jsx (hunk 4 above)
Comment (annemirasol) [reply client/my-sites/people/team-list/team.jsx]: Yeah, it's a tough one. I don't see other instances we can crib off of. We can go with it for now and perhaps ask the translations team later.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6616288

Hi @naxoc, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Invite members to your workspace and manage their access settings.`
- `Invite members and guests to this P2 and manage their access settings.`
- `People in %(sitename)s`
- `People in this P2`
- `New workspace members will be able to access all the P2s in this workspace`
- `You can also invite people to specific P2s so they join as guests.`
- `These are members and guests who have joined this P2`
- `Manage your workspace members to view all the people in the %(workspaceName)s.`
- `Manage workspace members`
- `You are inviting people to %(sitename)s`
- `If they aren't already members of %(workspaceName)s they'll be invited as guests.`
- `Invite to the workspace instead`
- `This user is a member of this P2 as a guest, but not a member of the workspace.`
- `There is %(numberPeople)d member in this workspace`
- `There is %(numberPeople)d member in this P2`
- `%(numberGuests)d guest`
- `There is %(numMembers)d member and %(guests)s in this P2`

Thank you in advance!

**Comment by naxoc** [comment]
# Translator screenshots

There is %(numberPeople)d member in this workspace
Invite members to your workspace and manage their access settings.
People in %(sitename)s
Screen Shot 2021-09-16 at 16 40 48

---

New workspace members will be able to access all the P2s in this workspace
You can also invite people to specific P2s so they join as guests.
Screen Shot 2021-09-16 at 16 43 32

---

This user is a member of this P2 as a guest, but not a member of the workspace.
Manage workspace members
Screen Shot 2021-09-16 at 16 47 00

---

You are inviting people to %(sitename)s
If they aren't already members of %(workspaceName)s they'll be invited as guests.
Invite to the workspace instead
Screen Shot 2021-09-16 at 16 51 36

---

Invite members and guests to this P2 and manage their access settings.
People in this P2
These are members and guests who have joined this P2
Manage your workspace members to view all the people in the %(workspaceName)s.
There is %(numberPeople)d member in this P2
%(numberGuests)d guest
There is %(numMembers)d member and %(guests)s in this P2
Screen Shot 2021-09-16 at 16 52 57

**Comment by dlind1** [comment]
We've received some comments from our translators that the string `There is %(numMembers)d member and %(guests)s in this P2` is difficult to translate since it looks like there is a missing word (guests) after `%(guests)s`. Would it be possible to add a translators comment (`comment` property in the options array for translate call) to explain that `%(guests)s` is a separate translated string which expands to `%(numberGuests)d guest`?

**Comment by naxoc** [comment]
Hi @dlind1 - yes absolutely. I added you as a reviewer on #56391

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6616288).

**Pull request merged**